### PR TITLE
NIFI-8528 Migrate NiFi Registry into NiFi codebase

### PR DESCRIPTION
**Pull request by tpalfy**
https://issues.apache.org/jira/browse/NIFI-8528

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by kevdoran** [comment]
Hi @tpalfy - thank you for this contribution. I have not had a chance to try it yet, but at a glance, I noticed a few items that are similar to @joewitt's feedback for #4933 regarding handling of LICENSE/NOTICE/README files to be updated to reflect that Registry, in this location, is a part of the the NiFi project/repo.

**Comment by gresockj** [comment]
Verified build and deployment on MacOS.  LGTM 👍

**Review by gresockj — APPROVED** [review]

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the work on this migration effort @tpalfy, as well as working through the build issues on Windows.  I am a bit concerned at the increased length of the build time with the introduction of the NiFi Registry and supporting modules.  The current NiFi builds have timed out frequently, so we may need to increase the default timeout from 90 minutes to two hours in the GitHub workflow configuration as part of this PR.

It may need to be a separate task, but the increased build time raises the question of more granular builds so that changes to NiFi Registry only result in building NiFi Registry.  Of course changes to core components will necessitate complete builds.  What do you think?

**Comment by tpalfy** [comment]
@exceptionfactory 
 > It may need to be a separate task, but the increased build time raises the question of more granular builds so that changes to NiFi Registry only result in building NiFi Registry. Of course changes to core components will necessitate complete builds. What do you think?

Yeah, the increased build time really feels to be a genuine issue. However I'm afraid we would want at least one full clean build no matter what, just to be sure.

Maybe it would work the other way around - _don't build registry_ if it hasn't change.
The reason being nifi depends on registry, not the other way around. So if the registry changes, we will need to build nifi for sure. Maybe this is not that apparent yet but there are plans to move some artifacts from registry to the _nifi-api_.

But I'm not sure what would be the best course of action.

**Comment by exceptionfactory** [comment]
> @exceptionfactory
> 
> > It may need to be a separate task, but the increased build time raises the question of more granular builds so that changes to NiFi Registry only result in building NiFi Registry. Of course changes to core components will necessitate complete builds. What do you think?
> 
> Yeah, the increased build time really feels to be a genuine issue. However I'm afraid we would want at least one full clean build no matter what, just to be sure.
> 
> Maybe it would work the other way around - _don't build registry_ if it hasn't change.
> The reason being nifi depends on registry, not the other way around. So if the registry changes, we will need to build nifi for sure. Maybe this is not that apparent yet but there are plans to move some artifacts from registry to the _nifi-api_.
> 
> But I'm not sure what would be the best course of action.

Thanks for the reply @tpalfy.  I agree that getting a successful build of your PR with Registry across all three environments sounds good.  Perhaps a follow-on issue to update the GitHub workflow configuration, including both timeout adjustments and selective builds, is the best way to go.

**Review by kevdoran — APPROVED** [review]
Nice work @tpalfy - thanks for making those requested changes to L&N.

This looks good and works well for me. Everything I could think to check in the port was addressed correctly - you did a good job spotting all the details to consider.

I am +1 with a one comment which could be addressed or disregarded at your discretion:

The NiFi Registry build installs Husky hooks for linting nifi-registry frontend code. 

Notice this during the build:

```
[INFO] husky > Setting up git hooks
[INFO] husky > Done
[INFO] husky > Like husky? You can support the project on Patreon:
[INFO] husky > https://www.patreon.com/typicode ❤
[INFO] added 2093 packages in 601.646s
```

And this result: 

```nifi git:NIFI-8528  ❯ ls -l .git/hooks/*
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/applypatch-msg
-rwxr-xr-x  1 kdoran  staff   478 Jan 21  2020 .git/hooks/applypatch-msg.sample
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/commit-msg
-rwxr-xr-x  1 kdoran  staff   896 Jan 21  2020 .git/hooks/commit-msg.sample
-rwxr-xr-x  1 kdoran  staff  3079 Jan 21  2020 .git/hooks/fsmonitor-watchman.sample
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/post-applypatch
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/post-checkout
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/post-commit
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/post-merge
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/post-receive
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/post-rewrite
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/post-update
-rwxr-xr-x  1 kdoran  staff   189 Jan 21  2020 .git/hooks/post-update.sample
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/pre-applypatch
-rwxr-xr-x  1 kdoran  staff   424 Jan 21  2020 .git/hooks/pre-applypatch.sample
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/pre-auto-gc
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/pre-commit
-rwxr-xr-x  1 kdoran  staff  1638 Jan 21  2020 .git/hooks/pre-commit.sample
-rwxr-xr-x  1 kdoran  staff   416 Jan 21  2020 .git/hooks/pre-merge-commit.sample
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/pre-push
-rwxr-xr-x  1 kdoran  staff  1348 Jan 21  2020 .git/hooks/pre-push.sample
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/pre-rebase
-rwxr-xr-x  1 kdoran  staff  4898 Jan 21  2020 .git/hooks/pre-rebase.sample
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/pre-receive
-rwxr-xr-x  1 kdoran  staff   544 Jan 21  2020 .git/hooks/pre-receive.sample
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/prepare-commit-msg
-rwxr-xr-x  1 kdoran  staff  1492 Jan 21  2020 .git/hooks/prepare-commit-msg.sample
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/push-to-checkout
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/sendemail-validate
-rwxr-xr-x  1 kdoran  staff  1256 May 21 13:40 .git/hooks/update
-rwxr-xr-x  1 kdoran  staff  3610 Jan 21  2020 .git/hooks/update.sample
```

These hooks affect every commit/push etc, and fail if host dependencies are not present. This may have made sense when registry was a standalone project, but as an optional component of this repository, most developers will be making changes to nifi and not registry, and these hooks should be disabled or opt-in rather than opt-out.

I do think this should be addressed, but other than that, I am +1 to merge this. The changes look good, and I tested NiFi Registry locally and in docker vis the nifi-registry-docker-maven output, and it worked great! Nice work!

**Comment by markap14** [comment]
Good catch @kevdoran, I would agree that we should get rid of the husky hooks. Looks there are a couple of conflicts that need to be cleaned up as well. I'd be a +1 otherwise, as well.

**Comment by tpalfy** [comment]
@kevdoran, @markap14  I agree, the Husky hooks should be opt-in.
That being said I'd rather have that done as a separate issue. I consider the change non-trivial that needs to be tested and confirmed by devs who have some experience with Husky. (I have not unfortunately.) Better not tie to this one imo.

File: nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/package.json [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    "husky": {
+        "hooks": {
+            "pre-commit": "lint-staged"
+        }
+    },
+    "lint-staged": {
+        "*.{js,jsx,ts,tsx}": "eslint",
+        "*.css": "stylelint",
+        "*.scss": "stylelint --syntax=scss"
+    },
```
Comment (exceptionfactory) [inline nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/package.json:46]: Removing these blocks along with husky from the `dependencies` section should avoid adding the Git hooks.

File: nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/package.json [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        "dtsgenerator": "2.0.6",
+        "eslint": "5.14.1",
+        "eslint-config-airbnb": "17.1.0",
+        "eslint-plugin-import": "2.16.0",
+        "eslint-plugin-jsx-a11y": "6.2.1",
+        "eslint-plugin-react": "7.12.4",
+        "file-loader": "3.0.1",
+        "html-loader": "0.5.5",
+        "html-webpack-plugin": "3.2.0",
+        "husky": "2.3.0",
```
Comment (exceptionfactory) [inline nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/package.json:104]: Removing this line will eliminate the husky dependency.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the work on the this effort @tpalfy!  I agree with @kevdoran and @markap14 that the Git hooks should be removed prior to merging this PR.  I noted two portions of the `package.json` that should eliminate the hooks.

**Review by kevdoran — APPROVED** [review]
+1, can verify the most recent commit results in husky hooks not being installed.

**Review by exceptionfactory — APPROVED** [review]

**Pull request merged**